src/Properties/properties.cpp

```diff
@@ -38,4 +39,114 @@ Vector<double, 3> calculateElectronicDipoleMoment(const VectorSQOneElectronOpera
 }
 
 
+/**
+ *  @param wavefunction1        a wave function in a product Fock space  
+ *  @param wavefunction2        a wave function in a product Fock space containing one fewer electron and the same amount of orbitals that is expressed in the same basis
+ *  
+ *  @return a vector with the coefficients of a Dyson orbital derived from the difference between the two wave functions expressed in the basis of the wave functions
+ */
+VectorX<double> calculateDysonAmplitudes(const WaveFunction& wavefunction1, const WaveFunction& wavefunction2) {
+
+    if (wavefunction1.get_fock_space().get_type() != FockSpaceType::ProductFockSpace) {
+        throw std::runtime_error("properties::calculateDysonOrbital(WaveFunction, WaveFunction): wavefunction1 is not in a product Fock space");
+    }
+
+    if (wavefunction2.get_fock_space().get_type() != FockSpaceType::ProductFockSpace) {
+        throw std::runtime_error("properties::calculateDysonOrbital(WaveFunction, WaveFunction): wavefunction2 is not in a product Fock space");
+    }
+
+    const auto fock_space1 = static_cast<const ProductFockSpace&>(wavefunction1.get_fock_space());
+    const auto fock_space2 = static_cast<const ProductFockSpace&>(wavefunction2.get_fock_space());
+
+    if (!((fock_space1.get_N_alpha() - fock_space2.get_N_alpha() == 0) && (fock_space1.get_N_beta() - fock_space2.get_N_beta() == 1))) {
```

Comment:
Maybe move this before casting the Fock spaces

Reply:
Can't access these variables before casting